honeycombio/resource_slo.go

```diff
@@ -39,9 +39,31 @@ func newSLO() *schema.Resource {
 			},
 			"dataset": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				ForceNew:    true,
 				Description: "The dataset this SLO is created in. Must be the same dataset as the SLI unless the SLI's dataset is `\"__all__\"`.",
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
```

Comment:
Thought about this a bit, and I like the pattern especially combined with the `getDataset` helper function as a way to drop `dataset` as a requirement for things that have environment-wide support -- and by extension remove the need to have users of the provider do `__all__` 😅 

Derived Columns most notably would benefit from `dataset` becoming optional, but I'm sure with a quick-ish audit I can find others.

To be clear: not for you to resolve here, but just saying I appreciate the thoughtful attention to the developer experience ✨